### PR TITLE
Add ability to favorite something

### PR DESCRIPTION
**Pull request by tlmurphy**
ezgif com-resize 3
Look at the commit lmao :trollface:

### CODE REVIEW

File: app/src/main/java/net/jmhossler/roastd/favoritestask/FavoritesActivity.java [hunk 1]
```diff
@@ -34,6 +29,6 @@ private void setupListFragment() {
       ActivityUtils.addFragmentToActivity(fm, silf, R.id.favorites_list_fragment_container);
     }
 
-    mFavoritesPresenter = new FavoritesPresenter(silf, FirebaseAuth.getInstance(), FirebaseRTSearchableItemRepository.getInstance(), FirebaseRTUserRepository.getsInstance());
```
Comment (jmhossler) [inline app/src/main/java/net/jmhossler/roastd/favoritestask/FavoritesActivity.java]: I still think that these things should be parameters to the favorites presenter

File: app/src/main/java/net/jmhossler/roastd/favoritestask/FavoritesPresenter.java [hunk 2]
```diff
@@ -19,19 +15,9 @@
 import static android.support.constraint.Constraints.TAG;
 
 public class FavoritesPresenter extends BaseSearchableItemPresenter {
-  private User mUser;
-  private UserDataSource mUserDataStore;
-  private SearchableItemDataSource mSearchableItemDataStore;
-  private FirebaseAuth mAuth;
 
-  public FavoritesPresenter(SearchableItemListContract.View v, FirebaseAuth firebaseAuth,
```
Comment (jmhossler) [inline app/src/main/java/net/jmhossler/roastd/favoritestask/FavoritesPresenter.java]: I don't think we should remove these parameters. We should pass them down to the super call.

File: app/src/main/java/net/jmhossler/roastd/viewtask/BaseSearchableItemPresenter.java [hunk 3]
```diff
@@ -1,22 +1,38 @@
...
+  public UserDataSource mUserDataStore;
+  public SearchableItemDataSource mSearchableItemDataStore;
+  public FirebaseAuth mAuth;
+  public User mUser;
 
   public BaseSearchableItemPresenter(SearchableItemListContract.View v) {
     mListView = v;
     mListView.setPresenter(this);
     mItems = new ArrayList<>();
+    mAuth = FirebaseAuth.getInstance();
```
Comment (jmhossler) [inline app/src/main/java/net/jmhossler/roastd/viewtask/BaseSearchableItemPresenter.java]: This makes testing harder. These should be passed down as parameters.

File: app/src/main/java/net/jmhossler/roastd/viewtask/BaseSearchableItemPresenter.java [hunk 4]
```diff
@@ -33,9 +49,50 @@ public void onListItemClicked(int position) {
     }
   }
 
+  @Override
+  public void onFavorite(int position) {
```
Comment (jmhossler) [inline app/src/main/java/net/jmhossler/roastd/viewtask/BaseSearchableItemPresenter.java]: should we have these as distinct functions (speaking of onFavorite and onUnfavorite), or should we have a toggleFavorite function that just calls toggleFavorite(position)?

File: app/src/main/java/net/jmhossler/roastd/viewtask/SearchableItemListFragment.java [hunk 5]
```diff
@@ -97,22 +100,40 @@ public void finish() {
...
                                 SearchableItemListContract.Presenter presenter) {
       super(inflater.inflate(R.layout.item_list_content, parent, false));
       mIconImageView = itemView.findViewById(R.id.list_item_icon);
       mContentView = itemView.findViewById(R.id.content);
+      mLikeButton = itemView.findViewById(R.id.heart_button);
       mPresenter = presenter;
       itemView.setOnClickListener(this);
+      mLikeButton.setOnLikeListener(new OnLikeListener() {
+        @Override
+        public void liked(LikeButton likeButton) {
```
Comment (jmhossler) [inline app/src/main/java/net/jmhossler/roastd/viewtask/SearchableItemListFragment.java]: I suppose this structure kind of lends itself to this, but I think toggle would still work and reduce the amount of code and similar logic.

**Comment by tlmurphy** [comment]
have mercy :trollface:

**Pull request merged**